polyfill/lib/monthday.mjs

```diff
@@ -98,10 +98,9 @@ export class MonthDay {
   }
   difference(other) {
     if (!ES.IsMonthDay(this)) throw new TypeError('invalid receiver');
-    other = ES.ToMonthDay(other);
     const [one, two] = [this, other].sort(MonthDay.compare);
```

Comment:
It seems like the `compare` method will probably catch any non-Temporal object, but I wonder if it wouldn't be better to explicitly brand-check in `difference` as well. (Same for other methods in this PR.)

Reply:
Indeed; did this in #343.